File: group8/day.java
//题目：输入某年某月某日，判断这一天是这一年的第几天？   
import java.util.*;
public class day {
public static void main(String[] args) {
     int year, month, day;
     int days = 0;
     int d = 0;
     int e;
     input fymd = new input();
     do {
     e = 0;
     System.out.print("输入年：");
   Scanner s = new Scanner(System.in);
     year = s.nextInt();
     System.out.print("输入月：");
   Scanner ss = new Scanner(System.in);
     month = ss.nextInt();
     System.out.print("输入天：");
   Scanner sss = new Scanner(System.in);
     day = sss.nextInt();
     if (year < 0 || month < 0 || month > 12 || day < 0 || day > 31) {
     System.out.println("输入错误，请重新输入！");
     e=1 ; 
     }
     }while( e==1);
      for (int i=1; i <month; i++) {
      switch (i) {
      case 1:
      case 3:
      case 5:
      case 7:
      case 8:
      case 10:
      case 12:
       days = 31;
      break;
      case 4:
      case 6:
      case 9:
      case 11:
       days = 30;
      break;
      case 2:
       if ((year % 400 == 0) || (year % 4 == 0 && year % 100 != 0)) {
        days = 29;
       } else {
        days = 28;
       }
       break;
      }
      d += days;
      }
     System.out.println(year + "-" + month + "-" + day + "是这年的第" + (d+day) + "天。");
}
}

File: group8/day.py
d = 0
while True:
    year = input('输入年：')
    year = int(year)
    month = input('输入月：')
    month = int(month)
    day = input('输入日：')
    day = int(day)
    e = 0
    i = 1
    if year < 0 or month < 0 or month > 12 or day < 0 or day > 31:
        e = 1
        print('输入错误，请重新输入！')
    if e == 0:
        break
if month == 1:
    print(year, '-', month, '-', day, '是这年的第', day, '天。')
else:
    for i in range(1, month):
        if i in (1, 3, 5, 7, 8, 10, 12):
            days = 31
        elif i in (4, 6, 9, 11):
            days = 30
        else:
            if year % 400 == 0 or (year % 4 == 0 and year % 100 != 0):
                days = 29
            else:
                days = 28
    d = d + days
    print(year, '-', month, '-', day, '是这年的第', (d + day), '天。')